File: Array/easy/2942. Find Words Containing Character/findWordsContaining.cpp
//time complexity : O(n)
//space complecity : O(n)
class Solution 
{
public:
    //in this solution i loop in the vector and look for branch
    vector<int> findWordsContaining(vector<string>& words, char x) 
    {
        vector<int> ans;
        for (int i = 0; i < words.size(); i++)
        {
            
            //when branch is true add index to the answer vector
            if (words[i].find(x) != string::npos) ans.push_back(i);
        }
        return ans;
    }
};

File: Array/easy/2942. Find Words Containing Character/findWordsContaining.py
#time complexity : O(n)
#space complecity : O(n)
class Solution(object):
    #in this solution i loop in the list and look for branch
    def findWordsContaining(self, words, x):
        ans = []
        for i in range(len(words)):
            if x in words[i]:
                #when branch is true add index to the answer list
                ans.append(i)
                
        return ans